### PR TITLE
Add soil model

### PR DESCRIPTION
**Pull request by jb-mackay**
Adds a standalone soil model draft conforming the the initial LSM interface

### CODE REVIEW

File: src/Soil.jl [hunk 1]
```diff
@@ -1,41 +1,164 @@
...
-    make_ode_function,
-    prognostic_vars,
-    auxiliary_vars
-export RichardsModel
-
-"""
-    RichardsModel
+    make_ode_function
+using UnPack
+export RichardsModel, RichardsParameters
```
Comment (kmdeck) [inline src/Soil.jl:15]: document exported items?

**Review by kmdeck — APPROVED** [review]
looks good, Ben! For this PR, mostly minor (can use the defaults in some cases, no need for new methods) + cleaning up commented notes if not needed anymore, adding a little documentation.

remove `z` from aux in this PR or a next one
add in other aux variables for testing in this PR or a next one

File: test/runtests.jl [hunk 2]
```diff
@@ -2,10 +2,8 @@ using Test
 
 using ClimaLSM
 using ClimaLSM.Domains: Column, RootDomain
-include(joinpath("../src", "Soil.jl"))
-using .Soil
-include(joinpath("../src", "Roots.jl"))
-using .Roots
+using ClimaLSM.Soil
```
[the remark this answers is not in the document]
Comment (jb-mackay) [reply test/runtests.jl:5]: Good point!

File: src/notes.txt [hunk 3]
```diff
@@ -0,0 +1,64 @@
+notes
```
[the remark this answers is not in the document]
Comment (jb-mackay) [reply src/notes.txt:1]: Good plan - thanks!

File: src/Soil.jl [hunk 4]
```diff
@@ -1,41 +1,164 @@
...
+
+function initialize_prognostic(
+    soil::RichardsModel,
+    coords::Union{ClimaCore.Fields.Field, Vector{FT}, FT},
+) where {FT}
+    init_prog(z) = (ϑ_l = similar(z),)
+    prognostic_state = ClimaCore.Fields.FieldVector(; soil = init_prog(coords))
+    return prognostic_state
+end
+### Should coords be stored here or within the model? Keep in mind when extending to other components and domains.
```
[the remark this answers is not in the document]
Comment (jb-mackay) [reply src/Soil.jl]: I think I agree with your other comment that lingering notes should be removed and turned into issues, if need be

File: src/Soil.jl [hunk 5]
```diff
@@ -1,41 +1,164 @@
...
+function initialize_auxiliary(
```
[the remark this answers is not in the document]
Comment (jb-mackay) [reply src/Soil.jl]: Yes, let's add this soon but in a future PR.

File: src/Soil.jl [hunk 6]
```diff
@@ -1,41 +1,130 @@
...
         param_set,
         domain,
+        coords,
+        boundary_exchanges,
         :soil,
     )
 end
 
-prognostic_vars(model::RichardsModel) = (:ϑ_l,)
+coordinates(model::RichardsModel) = model.coordinates
```
Comment (kmdeck) [inline src/Soil.jl:38]: awesome! this + the model constructor nicely prevents the user from calling coordinates(domain::Column) multiple times and ending up with coordinates from different spaces.

**Pull request merged**